Added example code which demonstrates the build tools and testing wit…

dhwitz opened the pull request:
Contains the structure that will be used for the project.

- Gradle files
- gitignore
- Sample code for both code to run error prone on and a sample plugin module
- Tests for sample code

lukhnos commented on plugin/build.gradle:
```diff
@@ -0,0 +1,15 @@
+  annotationProcessor 'com.google.auto.service:auto-service:1.0-rc4'
+}
+
+//configurations {
```
Are these comments from the Error Prone sample code?

dhwitz replied:
At one point, I could not get the tests to build without that part. I commented it out because it was fixed by something else I did. I can keep a note of it somewhere else in case I need it in the future and remove it from the code.

dhwitz commented:
I agree the `.idea` directory can be skipped. When I was looking online, some people were concerned that even with Gradle files there could be issues with importing. I just tested it and it works fine without it. I have removed it from the project and added `.idea` to the gitignore.
